### PR TITLE
[coverage] Ignore compiler_crashers for coverage tests

### PR DESCRIPTION
**Pull request by harlanhaskins**
#### What's in this pull request?

Ignores `compiler_crashers` when profiling coverage.
1) Compiler crashers should not count for coverage; it's not well known which parts of the compiler each of them touches, nor do we intend to use those tests to find that out.
2) The compiler crashers run very quickly and fill up disk space before the merger has time to merge them. This is a stopgap solution until we can have `lit` communicate with the merger better, or until LLVM can handle in-process merging.

#### Resolved bug number: N/A

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please test

File: test/lit.cfg [hunk 1]
```diff
@@ -147,20 +148,27 @@ class SwiftTest(lit.formats.ShTest, object):
     def before_test(self, test, litConfig):
         if self.coverage_mode:
             profdir = self.profdir_for_test(test)
+            # FIXME: The compiler crashers run so fast they fill up the
+            # merger's queue (and therefore the build bot's disk)
+            if 'compiler_crashers' in profdir:
```
Comment (gribozavr) [inline test/lit.cfg]: Do you also want to cover `validation-test/SIL/crashers` and `validation-test/SIL/crashers_fixed`?

File: test/lit.cfg [hunk 2]
```diff
@@ -139,6 +139,7 @@ class SwiftTest(lit.formats.ShTest, object):
             self.coverage_mode = None
         else:
             self.coverage_mode = coverage_mode
+        self.skip_merge_next_test = False
```
Comment (gribozavr) [inline test/lit.cfg]: Could you invert the semantics of this option to be `merge_next_test`?  Double-negations are harder to reason about.

File: test/lit.cfg [hunk 3]
```diff
@@ -147,20 +148,27 @@ class SwiftTest(lit.formats.ShTest, object):
...
+                return
             if not os.path.exists(profdir):
                 os.makedirs(profdir)
 
             test.config.environment["LLVM_PROFILE_FILE"] = \
                 os.path.join(profdir, "swift-%p.profraw")
 
     def after_test(self, test, litConfig, result):
-        if self.coverage_mode == "MERGED":
+        if self.coverage_mode == "MERGED" and not self.skip_merge_next_test:
```
Comment (gribozavr) [inline test/lit.cfg]: Are `before_test`/`after_test` run without interleaving for different tests?

File: test/lit.cfg (hunk 1 above)
Comment (harlanhaskins) [inline test/lit.cfg]: I suppose we should; should I change this to `if 'crashers' in profdir`?

Comment (gribozavr) [inline test/lit.cfg]: That should work.

**Comment by harlanhaskins** [comment]
@gribozavr I reworked how I detect and keep track of crashes; that should address the concerns you had. Thanks for the fresh set of eyes!

**Comment by gribozavr** [comment]
This makes a lot of sense to me, LGTM!

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by harlanhaskins** [comment]
@swift-CI please Python lint

**Pull request merged**